### PR TITLE
Implement first version of profiler

### PR DESCRIPTION
**Pull request by DrRataplan**

### CODE REVIEW

File: src/performance.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	performance.mark(xpath);
+}
+export function markXPathEnd(xpath: string) {
+	if (!profilingEnabled) {
+		return;
+	}
+	// Replace the mark with a measure of the time spent
+	performance.measure(`xpath: ${xpath}`, xpath, undefined);
+	performance.clearMarks(xpath);
```
Comment (JosVerburg) [inline src/performance.ts:161]: Will this break if I call the same xpath function recursively twice? For this I could register a custom xpath function which keeps executing itself until an external value changes.

**Review by JosVerburg — APPROVED** [review]
Looks nice. Might be worth to also add some of the explanation to the README.md to make the feature easier to find. Could put it under the ##debug chapter or so.

File: src/performance.ts (hunk 1 above)
Comment (DrRataplan) [reply src/performance.ts]: Fixed by adding a counter

**Comment by DrRataplan** [comment]
Add tools for performance profiling

See README on how to use it

**Pull request merged**